tests/saitests/py3/sai_qos_tests.py

```diff
@@ -1957,60 +2183,70 @@ def runTest(self):
         else:
             cell_occupancy = 1
 
-        pkt_dst_mac2 = router_mac if router_mac != '' else dst_port_2_mac
         pkt_dst_mac3 = router_mac if router_mac != '' else dst_port_3_mac
 
         is_dualtor = self.test_params.get('is_dualtor', False)
         def_vlan_mac = self.test_params.get('def_vlan_mac', None)
         if is_dualtor and def_vlan_mac is not None:
             pkt_dst_mac = def_vlan_mac
-            pkt_dst_mac2 = def_vlan_mac
             pkt_dst_mac3 = def_vlan_mac
 
-        pkt = construct_ip_pkt(packet_length,
-                               pkt_dst_mac,
-                               src_port_mac,
-                               src_port_ip,
-                               dst_port_ip,
-                               dscp,
-                               src_port_vlan,
-                               ecn=ecn,
-                               ttl=ttl)
-        dst_port_id = self.get_rx_port(
-            src_port_id, pkt_dst_mac, dst_port_ip, src_port_ip, dst_port_id, src_port_vlan
-        )
+        pkt_s = get_multiple_flows(
```

Comment:
@rraghav-cisco We've encountered infinite loops here in the past and decided to use get_multiple_flows() only for cisco platforms.
This feature is now available on all platforms if required. More testing on multiple platforms is needed.

@bingwang-ms what is your comments?

Reply:
@XuChen-MSFT : This issue is fixed and verified to work. We are now limiting the loop to only 20 attempts, and failing which, we just pick the interface where the packets are coming anyway, instead of the dst_id.